### PR TITLE
JAudio2 Work

### PR DESCRIPTION
**Pull request by jdflyer**

### CODE REVIEW

File: include/JSystem/JAudio2/JAISound.h [hunk 1]
```diff
@@ -10,17 +10,23 @@
 
 class JAISoundID {
 public:
-    operator u32() const { return this->mId; }
-    void operator=(JAISoundID const& other) {mId = other.mId;};
+    operator u32() const { return this->mId.mId; }
```
Comment (TakaRikka) [inline include/JSystem/JAudio2/JAISound.h:13]: is there a better name to give one of these vars? `mId.mId` is kinda confusing to read and understand what the code is doing

File: include/JSystem/JAudio2/JASWaveArcLoader.h [hunk 2]
```diff
@@ -1,6 +1,109 @@
...
+#include "MSL_C/MSL_Common/Src/string.h"
+#include "dolphin/os/OSMutex.h"
+
+#include "JSystem/JAudio2/JASDvdThread.h"
+#include "JSystem/JAudio2/JASHeapCtrl.h"
+#include "JSystem/JKernel/JKRDvdAramRipper.h"
+#include "JSystem/JKernel/JKRSolidHeap.h"
+
+struct JASDisposer {
+    JASDisposer(){};
```
Comment (TakaRikka) [inline include/JSystem/JAudio2/JASWaveArcLoader.h:17]: comma after braces is unneeded and space after parenthesis

File: include/JSystem/JAudio2/JASWaveArcLoader.h [hunk 3]
```diff
@@ -1,6 +1,109 @@
...
+    /* 802903F4 */ bool allocTail(JASHeap*, u32);
+    /* 802904E4 */ void free();
+    /* 80290608 */ void insertChild(JASHeap*, JASHeap*, void*, u32, bool);
+    /* 802906F0 */ void getTailHeap();
+    /* 8029077C */ void getTailOffset();
+    /* 802907E0 */ void getCurOffset();
+    /* 80290B54 */ ~JASHeap(){};
+    void* getBase() { return mBase; }
+
+    /*0x1c*/ OSMutex mMutex;
```
Comment (TakaRikka) [inline include/JSystem/JAudio2/JASWaveArcLoader.h:35]: space at comment start / end and capitalize hex numbers

File: include/JSystem/JAudio2/JAUStreamFileTable.h [hunk 4]
```diff
@@ -1,6 +1,32 @@
...
+    /* 0x0 */ const BinaryStreamFileTable* mData;
+};
+
+struct JAIStreamDataMgr {
+    /* 802A3AD8 */ ~JAIStreamDataMgr();
+};
+
+struct JAUStreamDataMgr_StreamFileTable : JAIStreamDataMgr {
+    /* 802A74AC */ virtual int getStreamFileEntry(JAISoundID);
+    /* 802A74E8 */ virtual ~JAUStreamDataMgr_StreamFileTable(){};
```
Comment (TakaRikka) [inline include/JSystem/JAudio2/JAUStreamFileTable.h:27]: parenthesis space and unnecessary comma

File: src/f_op/f_op_scene.cpp [hunk 5]
```diff
@@ -20,8 +20,8 @@ static void fopScn_Execute(scene_class* pScene) {
 }
 
 /* 8001EB84-8001EBAC 0194C4 0028+00 1/0 0/0 0/0 .text            fopScn_IsDelete__FPv */
-static s32 fopScn_IsDelete(scene_class* pScene) {
-    return fpcMtd_IsDelete(pScene->mpMtd, pScene);
+static s32 fopScn_IsDelete(void* pScene) {
+    return fpcMtd_IsDelete(((scene_class*)pScene)->mpMtd, pScene);
```
Comment (TakaRikka) [inline src/f_op/f_op_scene.cpp:24]: a `static_cast` might look cleaner here

**Review by TakaRikka — APPROVED** [review]

**Review by notyourav — APPROVED** [review]

**Pull request merged**